File: app/src/main/java/no/joharei/flixr/network/models/Photos.java
package no.joharei.flixr.network.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

@SuppressWarnings("unused")
class Photos {

    private String id;
    private String primary;
    private String owner;
    @SerializedName("ownername")
    private String ownerName;
    @SerializedName("photo")
    private List<Photo> photos = new ArrayList<>();
    private int page;
    @SerializedName("perpage")
    private int perPage;
    private int pages;
    private String total;
    private String title;

    List<Photo> getPhotos() {
        return photos;
    }
}

File: app/src/main/java/no/joharei/flixr/api/models/Photos.kt
package no.joharei.flixr.api.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PhotosResponse(val photoset: Photos)

@JsonClass(generateAdapter = true)
data class Photos(
    val id: String,
    val primary: String,
    val owner: String,
    val ownername: String,
    @Json(name = "photo")
    val photos: List<Photo>,
    val page: Int,
    val per_page: Int,
    val pages: Int,
    val total: String,
    val title: String
)